client/my-sites/sharing/controller.js

```diff
@@ -1,97 +1,83 @@
-			} ),
-			document.getElementById( 'primary' ),
-			context.store
-		);
-	},
-
-	connections: function( context, next ) {
-		var SharingConnections = require( 'my-sites/sharing/connections/connections' ),
-			site = sites.getSelectedSite(),
-			basePath = route.sectionify( context.path ),
-			baseAnalyticsPath;
-
-		if ( site ) {
-			baseAnalyticsPath = basePath + '/:site';
-		} else {
-			baseAnalyticsPath = basePath;
-		}
-
-		if ( site && ! utils.userCan( 'publish_posts', site ) ) {
-			notices.error( i18n.translate( 'You are not authorized to manage sharing settings for this site.' ) );
-		}
-
-		if ( site && site.jetpack && ! site.isModuleActive( 'publicize' ) ) {
-			// Redirect to sharing buttons if Jetpack Publicize module is not
-			// active, but ShareDaddy is active
-			page.redirect( site.isModuleActive( 'sharedaddy' ) ? '/sharing/buttons/' + sites.selected : '/stats' );
-		} else {
-			analytics.pageView.record( baseAnalyticsPath, analyticsPageTitle + ' > Connections' );
-
-			context.contentComponent = React.createElement( SharingConnections );
-		}
-
-		next();
-	},
-
-	buttons: function( context, next ) {
-		var SharingButtons = require( 'my-sites/sharing/buttons/buttons' ),
-			site = sites.getSelectedSite(),
-			basePath = route.sectionify( context.path ),
-			baseAnalyticsPath;
-
-		if ( site ) {
-			baseAnalyticsPath = basePath + '/:site';
-		} else {
-			baseAnalyticsPath = basePath;
-		}
-
-		analytics.pageView.record( baseAnalyticsPath, analyticsPageTitle + ' > Sharing Buttons' );
-
-		if ( site && ! utils.userCan( 'manage_options', site ) ) {
-			notices.error( i18n.translate( 'You are not authorized to manage sharing settings for this site.' ) );
-		}
-
-		if ( site && site.jetpack && ( ! site.isModuleActive( 'sharedaddy' ) || site.versionCompare( '3.4-dev', '<' ) ) ) {
-			notices.error( i18n.translate( 'This page is only available to Jetpack sites running version 3.4 or higher with the Sharing module activated.' ) );
-		}
-
-		context.contentComponent = React.createElement( SharingButtons );
-
-		next();
+		context.contentComponent = createElement( SharingConnections );
 	}
+
+	next();
+};
+
+export const buttons = ( context, next ) => {
+	const site = sites().getSelectedSite();
+	const basePath = sectionify( context.path );
+	const baseAnalyticsPath = site ? basePath + '/:site' : basePath;
+
+	pageView.record( baseAnalyticsPath, analyticsPageTitle + ' > Sharing Buttons' );
+
+	if ( site && ! utils.userCan( 'manage_options', site ) ) {
+		notices.error( translate( 'You are not authorized to manage sharing settings for this site.' ) );
+	}
+
+	if ( site && site.jetpack && ( ! site.isModuleActive( 'sharedaddy' ) || site.versionCompare( '3.4-dev', '<' ) ) ) {
+		notices.error( translate( 'This page is only available to Jetpack sites running version 3.4 or higher with the Sharing module activated.' ) );
+	}
+
+	context.contentComponent = createElement( SharingButtons );
+
+	next();
+};
+
+export default {
```

Comment:
Do we need the default export if we're already exporting the individual functions (appears `sharing/index.js` should work without the default export).